namer/consul/src/main/scala/io/buoyant/namer/consul/CatalogNamer.scala

```diff
@@ -108,30 +122,43 @@ class CatalogNamer(
       updateableAddr() = Addr.Pending
     }
 
-    def init(): Future[Unit] = mkRequest().flatMap(update).handle(handleUnexpected)
+    private[this] def metaFuture() =
```

Comment:
I don't think this refactor is necessary.  You should be able to just add a call to domainFuture in update (if setHost is set, of course).